nixos/modules/services/networking/dsnet.nix

```diff
@@ -0,0 +1,364 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+let
+  cfg = config.services.dsnet;
+  settingsFormat = pkgs.formats.json { };
+  patchFile = settingsFormat.generate "dsnet-patch.json" cfg.settings;
+in
+{
+  options.services.dsnet = {
+    enable = lib.mkOption {
+      type = lib.types.bool;
+      default = false;
+      description = "Enable dsnet as a systemd service.";
+    };
+
+    package = lib.mkOption {
+      type = lib.types.package;
+      default = (pkgs.callPackage ./dsnet.nix { });
```

Comment:
Huh? That can't be right. Also - you should just use `lib.mkPackageOption pkgs "dsnet" { }`